manticore/core/smtlib/solver.py

```diff
@@ -330,8 +330,7 @@ def can_be_true(self, constraints, expression):
 
         with constraints as temp_cs:
             temp_cs.add(expression)
-
-            self._reset(temp_cs.related_to(expression))
```

Comment:
If the expression has no variables, then it should be simplificable to a Constant. That's True or False.
In the False case the other constrains do not matter and the check should return False.
In the True case we'd need to check if the previous constraints set is feasible.
Something similar is fixed here: https://github.com/trailofbits/manticore/blob/dev-evm-arraystorage/manticore/core/smtlib/solver.py#L321-L335

Reply:
the changes mentioned in the other branch do not fix the same issue that this fixes, so i will keep these changes. for more see https://github.com/trailofbits/manticore/pull/787#issuecomment-371642163